scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -367,6 +369,21 @@ class ScioContext private[scio] (val options: PipelineOptions,
   }
 
   /**
+   * Get an SCollection for a Protobuf file.
+   * @group input
+   */
+  def protobufFile[T: ClassTag](path: String)
+                               (implicit ev: T <:< GeneratedMessage with Message): SCollection[T] =
+    pipelineOp {
+      // protobuf parser is not serializable - defeat via twitter's externalizer
+      val serializableParser = Externalizer(ScioUtil.getProtobufParser[T])
+      this.objectFile[Array[Byte]](path).map{ bytes =>
+        lazy val parser = serializableParser.get
```

Comment:
is there a better way to do this? we could potentially add something like AvroIO (ProtobufIO) that would handle parser and process elements as they come. @nevillelyh ?